pkg/apiserver/registry/security/workloadidentity/storage/jwt.go

```diff
@@ -68,87 +81,165 @@ func (r *TokenRequestREST) getGardenerClaims(workloadIdentity *securityapi.Workl
-		project *gardencorev1beta1.Project
-
-		shootMeta, seedMeta, projectMeta metav1.Object
-		err                              error
-		coreInformers                    = r.coreInformerFactory.Core().V1beta1()
+		ctxObjects   = &contextObjects{}
+		shoot        *gardencorev1beta1.Shoot
+		seed         *gardencorev1beta1.Seed
+		project      *gardencorev1beta1.Project
+		backupBucket *gardencorev1beta1.BackupBucket
+		backupEntry  *gardencorev1beta1.BackupEntry
+
+		err error
 	)
 
 	switch gvk := schema.FromAPIVersionAndKind(ctxObj.APIVersion, ctxObj.Kind); {
 	case gvk.Group == gardencorev1beta1.SchemeGroupVersion.Group && gvk.Kind == "Shoot":
-		if shoot, err = coreInformers.Shoots().Lister().Shoots(*ctxObj.Namespace).Get(ctxObj.Name); err != nil {
-			return nil, nil, nil, err
+		if shoot, err = r.shootListers.Shoots(*ctxObj.Namespace).Get(ctxObj.Name); err != nil {
+			return nil, err
 		}
-		shootMeta = shoot.GetObjectMeta()
+		ctxObjects.shoot = shoot.GetObjectMeta()
 
 		if shoot.UID != ctxObj.UID {
-			return nil, nil, nil, fmt.Errorf("uid of contextObject (%s) and real world resource(%s) differ", ctxObj.UID, shoot.UID)
+			return nil, fmt.Errorf("uid of contextObject (%s) and real world resource(%s) differ", ctxObj.UID, shoot.UID)
 		}
 
 		if shoot.Spec.SeedName != nil {
 			seedName := *shoot.Spec.SeedName
-			if seed, err = coreInformers.Seeds().Lister().Get(seedName); err != nil {
-				return nil, nil, nil, err
+			if seed, err = r.seedLister.Get(seedName); err != nil {
+				return nil, err
 			}
-			seedMeta = seed.GetObjectMeta()
+			ctxObjects.seed = seed.GetObjectMeta()
 		}
 
-		if project, err = admissionutils.ProjectForNamespaceFromLister(coreInformers.Projects().Lister(), shoot.Namespace); err != nil {
-			return nil, nil, nil, err
+		if project, err = admissionutils.ProjectForNamespaceFromLister(r.projectLister, shoot.Namespace); err != nil {
+			return nil, err
 		}
-		projectMeta = project.GetObjectMeta()
+		ctxObjects.project = project.GetObjectMeta()
 
 	case gvk.Group == gardencorev1beta1.SchemeGroupVersion.Group && gvk.Kind == "Seed":
-		if seed, err = coreInformers.Seeds().Lister().Get(ctxObj.Name); err != nil {
-			return nil, nil, nil, err
+		if seed, err = r.seedLister.Get(ctxObj.Name); err != nil {
+			return nil, err
 		}
 
 		if seed.UID != ctxObj.UID {
-			return nil, nil, nil, fmt.Errorf("uid of contextObject (%s) and real world resource(%s) differ", ctxObj.UID, seed.UID)
+			return nil, fmt.Errorf("uid of contextObject (%s) and real world resource(%s) differ", ctxObj.UID, seed.UID)
+		}
+		ctxObjects.seed = seed.GetObjectMeta()
+
+	case gvk.Group == gardencorev1beta1.SchemeGroupVersion.Group && gvk.Kind == "BackupBucket":
+		if backupBucket, err = r.backupBucketLister.Get(ctxObj.Name); err != nil {
+			return nil, err
+		}
+
+		if backupBucket.UID != ctxObj.UID {
+			return nil, fmt.Errorf("uid of contextObject (%s) and real world resource(%s) differ", ctxObj.UID, backupBucket.UID)
```

Comment:
Nit (also for other occurrences):
I wonder if we should make the error message more explicit about the type of objects. I imagine that only UIDs would be hard to find without additional context.

```suggestion
			return nil, fmt.Errorf("uid of contextObject (%s) and real world resource (%s) differ", ctxObj.UID, backupBucket.UID)
```

Reply:
The error is associated with the secret being reconciled, that should be enough to identify the context object. 
It is something like 
```json
{
    "Secret": {
        "name": "bucket-088001bb-6462-4212-bfdb-af0bb8f6db05",
        "namespace": "garden"
    },
    "namespace": "garden",
    "name": "bucket-088001bb-6462-4212-bfdb-af0bb8f6db05",
    "error": "could not request token for WorkloadIdentity: failed to resolve context object: uid of contextObject (c867e761-2b5a-4d7f-8af3-1b0ee7d2d3be) and real world resource(c867e761-2b5a-4d7f-8af3-1b0ee7d2d3bf) differ"
    ...,
}
```

As this error message is exactly the same for all context objects, I would just add the kind of object, i.e.


```suggestion
			return nil, fmt.Errorf("uid of contextObject (%s) and real world BackupBucket resource (%s) differ", ctxObj.UID, backupBucket.UID)
```

wdyt?

Reply:
Yes, sound good

Reply:
Done with 1046ffcd0f7add618af9c1b76aaa7cef1ae90b39